### PR TITLE
upgrade to ruby 2.7.1

### PR DESCRIPTION
**Pull request by machisuji**

### CODE REVIEW

**Comment by ulferts** [comment]
Please also take a look at the documentation e.g. https://github.com/opf/openproject/blob/dev/docs/development/development-environment-ubuntu/README.md and https://github.com/opf/openproject/blob/dev/docs/installation-and-operations/installation/manual/README.md
We seem to tend to forget those places.

**Review by ulferts — CHANGES_REQUESTED** [review]
That ruby bump seems to be pretty straight forward. 

The failing feature specs fail because of some frozen string modification which shouldn't be to hard to fix. 

But even with those fixed, there is still the issue of the multitude of deprecation warnings, a lot of which do not stem from our code. We could of course disable them but if we do, we should at least have the deprecations in our code fixed to not be surprised later on.

File: docs/development/development-environment-osx/README.md [hunk 1]
```diff
@@ -31,21 +31,21 @@ $ rbenv init
 
 **Installing ruby-2.6**
```
Comment (akasparas) [inline docs/development/development-environment-osx/README.md]: Old version still slipped in the heading

**Comment by machisuji** [comment]
The deprecation warnings immediately visible outside of our own code are:

- rubytree
- awesome_nested_set
- git
- activerecord, actionpack, railties (well rails essentially)

For the last one a RC is out which should fix the warnings. I don't expect any fixes any time soon for the first two gems, though, as they haven't been updated in a while. We could go ahead and fix the issues ourselves and propose them, I reckon.

The deprecation in git is fixed with an update to 1.7.0 which I included now.
We may want to wait until the official rails update at least until doing the upgrade.

**Comment by machisuji** [comment]
A rails update to 6.0.3 resolved some more warnings. But railties and activesupport still use some deprecated features (Passing the keyword argument as the last hash parameter) in some places.

**Comment by ulferts** [comment]
Thanks for staying on the topic @machisuji. I'll see if I can find a solution for the `URI.escape is obsolete` warning which seems to be all over the place. 
We should probably cherry-pick the rails bump into dev to benefit from it right away as it might still take some time for us to get this PR merged.

**Comment by ulferts** [comment]
I cherry-picked your commit https://github.com/opf/openproject/pull/8327/commits/54943f50736fa9f01ef271d0570a5ae3f1cfdc41 to a rails bump branch (#8381)

**Review by ulferts — APPROVED** [review]
Good to merge from my side now. There are still a whole lot of deprecations but those we can address separately.

**Pull request merged**